src/OverlayDockContainer.cpp

```diff
@@ -0,0 +1,315 @@
+/*******************************************************************************
+** Qt Advanced Docking System
+** Copyright (C) 2017 Uwe Kindler
+**
+** This library is free software; you can redistribute it and/or
+** modify it under the terms of the GNU Lesser General Public
+** License as published by the Free Software Foundation; either
+** version 2.1 of the License, or (at your option) any later version.
+**
+** This library is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+** Lesser General Public License for more details.
+**
+** You should have received a copy of the GNU Lesser General Public
+** License along with this library; If not, see <http://www.gnu.org/licenses/>.
+******************************************************************************/
+
+
+//============================================================================
+/// \file   DockWidgetTab.h
+/// \author Syarif Fakhri
+/// \date   05.09.2022
+/// \brief  Implementation of COverlayDockContainer class
+//============================================================================
+
+//============================================================================
+//                                   INCLUDES
+//============================================================================
+#include "OverlayDockContainer.h"
+#include "DockManager.h"
+#include "DockWidgetSideTab.h"
+#include "DockWidgetTab.h"
+#include "SideTabBar.h"
+#include "DockAreaTitleBar.h"
+#include "DockAreaWidget.h"
+#include "DockingStateReader.h"
+
+#include <QXmlStreamWriter>
+#include <QAbstractButton>
+#include <QBoxLayout>
+#include <QPainter>
+#include <QSplitter>
+
```

Comment:
Compiling this files causes error: **OverlayDockContainer.cpp:52:32: error: field 'DockManager' has incomplete type 'QPointer<ads::CDockManager>'**

Add QPointer include:

```cpp
#include <QPointer>
```